public/static-checkout/js/scripts.js

```diff
@@ -0,0 +1,313 @@
+/* eslint-disable vars-on-top */
+/* eslint-disable prefer-template */
+/* eslint-disable prefer-arrow-callback */
+/* eslint-disable no-var */
+var checkoutForm = document.getElementById('checkout-form');
+var paypalButton = document.getElementById('submit-paypal');
+var couponButton = document.getElementById('coupon-button');
+var userButton = document.querySelector('.user-button');
+var countrySelect = document.getElementById('country');
+var stateSelect = document.getElementById('state');
+var countryStates = {
+  CA: {
+    AB: 'Alberta',
+    BC: 'British Columbia',
+    MB: 'Manitoba',
+    NB: 'New Brunswick',
+    NL: 'Newfoundland and Labrador',
+    NS: 'Nova Scotia',
+    NT: 'Northwest Territories',
+    NU: 'Nunavut',
+    ON: 'Ontario',
+    PE: 'Prince Edward Island',
+    QC: 'Québec',
+    SK: 'Saskatchewan',
+    YT: 'Yukon',
+  },
+  US: {
+    AL: 'Alabama',
+    AK: 'Alaska',
+    AZ: 'Arizona',
+    AR: 'Arkansas',
+    AA: 'Armed Forces Americas',
+    AE: 'Armed Forces Europe',
+    AP: 'Armed Forces Pacific',
+    CA: 'California',
+    CO: 'Colorado',
+    CT: 'Connecticut',
+    DE: 'Delaware',
+    DC: 'District Of Columbia',
+    FL: 'Florida',
+    GA: 'Georgia',
+    HI: 'Hawaii',
+    ID: 'Idaho',
+    IL: 'Illinois',
+    IN: 'Indiana',
+    IA: 'Iowa',
+    KS: 'Kansas',
+    KY: 'Kentucky',
+    LA: 'Louisiana',
+    ME: 'Maine',
+    MD: 'Maryland',
+    MA: 'Massachusetts',
+    MI: 'Michigan',
+    MN: 'Minnesota',
+    MS: 'Mississippi',
+    MO: 'Missouri',
+    MT: 'Montana',
+    NE: 'Nebraska',
+    NV: 'Nevada',
+    NH: 'New Hampshire',
+    NJ: 'New Jersey',
+    NM: 'New Mexico',
+    NY: 'New York',
+    NC: 'North Carolina',
+    ND: 'North Dakota',
+    OH: 'Ohio',
+    OK: 'Oklahoma',
+    OR: 'Oregon',
+    PA: 'Pennsylvania',
+    RI: 'Rhode Island',
+    SC: 'South Carolina',
+    SD: 'South Dakota',
+    TN: 'Tennessee',
+    TX: 'Texas',
+    UT: 'Utah',
+    VT: 'Vermont',
+    VA: 'Virginia',
+    WA: 'Washington',
+    WV: 'West Virginia',
+    WI: 'Wisconsin',
+    WY: 'Wyoming',
+  },
+};
+var ecommerceBaseUrl;
+if (window.location.hostname === 'payment.edx.org') {
+  ecommerceBaseUrl = 'https://ecommerce.edx.org';
+} else if (window.location.hostname === 'payment.stage.edx.org') {
+  ecommerceBaseUrl = 'https://ecommerce.stage.edx.org';
+} else if (window.location.hostname === 'localhost') {
+  ecommerceBaseUrl = 'http://localhost:18130';
+}
+
+userButton.addEventListener('click', function () {
+  document.querySelector('.dropdown-menu').classList.toggle('show');
+}, false);
+
+window.onclick = function closeMenu(event) {
+  if (!event.target.matches('.user-button')) {
+    var dropdownMenu = document.getElementsByClassName('dropdown-menu');
+    for (var i = 0; i < dropdownMenu.length; i++) {
+      var open = dropdownMenu[i];
+      if (open.classList.contains('show')) {
+        open.classList.remove('show');
+      }
+    }
+  }
+};
+
+function redirectToMFE(couponCode) {
```

Comment:
I imagine we will want to exclude learners who get redirected to the MFE from the experiment and we will want to exclude them from the routing as well, so we probably want to add a query parameter here to reflect that we are coming from the static checkout page for the routing code to look at.

Reply:
hmm... @inventhouse would we know this information at analysis time because we see them on both pages?

Reply:
queries like that aren't trivial to write in the analysis, but regardless we will need a parameter to exclude those users from the routing, otherwise they could get routed right back to the static checkout page